src/edges_analysis/analysis/beams.py

```diff
@@ -260,6 +260,101 @@ def from_feko(cls, path: [str, Path], az_antenna_axis: float = 0) -> Beam:
             raw_file=path,
         )
 
+    @classmethod
+    def from_feko_raw(
+        cls,
+        file_name_prefix: [str, Path],
+        ext: str = "txt",
+        f_low: int = 40,
+        f_high: int = 100,
+        freq_p: int = 61,
+        theta_p: float = 181,
+        phi_p: float = 361,
+        az_antenna_axis: float = 0,
+    ) -> Beam:
+        """
+        Read a FEKO beam file.
+
+        Parameters
+        ----------
+        filename
+            The path to the file.
+        az_antenna_axis
+            The azimuth of the primary antenna axis, in degrees.
+
+        Returns
+        -------
+        beam_maps
+            A ``(Nfreq, Nel, Naz)`` array giving values of the beam. Note that elevation
+            and azimuth are always in 1-degree increments.
+        freq
+            The frequencies at which the beam is defined.
+        """
+        beam_square = np.zeros((freq_p, theta_p, phi_p))
+        frequency = np.linspace(f_low, f_high, freq_p)
+        f1 = open(file_name_prefix + "_0-90." + ext)
+        f2 = open(file_name_prefix + "_91-180." + ext)
+        f3 = open(file_name_prefix + "_181-270." + ext)
+        f4 = open(file_name_prefix + "_271-360." + ext)
+
+        z = 181 * 91 + 10  # ---> change this to no.of theta * no.of phi + No.of header lines
+        c = 0
+        for line in f1:
+            if c % z == 0:
+                co = 0
+            if c % z >= 10:
+                x = list(map(float, line.split()))
+                beam_square[int(c / z), co % 181, int(co / 181)] = 10 ** (x[8] / 10)
+                co = co + 1
+            c = c + 1
+
+        z = 181 * 90 + 10  #
+        c = 0
```

Comment:
try stick away from single-letter variables unless there's really good reason. Here, it'd be better to use something like

```python
for indx, line in enumerate(f2):
   ...
```

Then change all the c's to `indx`.

Reply:
And the same for f1